src/XMakeBuildEngine/Evaluation/ConditionEvaluator.cs

```diff
@@ -64,52 +65,55 @@ internal static void UpdateConditionedPropertiesTable
                     bool lastPiece = pieceSeparator < 0;
                     int pieceEnd = lastPiece ? leftValue.Length : pieceSeparator;
 
-                    Match singlePropertyMatch = s_singlePropertyRegex.Match(leftValue, pieceStart, pieceEnd - pieceStart);
-
-                    if (singlePropertyMatch.Success)
+                    if (leftValue.StartsWith("$("))
```

Comment:
Does this have a noticeable impact? I want to believe that the regex engine can do it just as fast.

If it does, please comment that it's an optimization with some justification.